L10-4　ChatSpace、投稿機能の非同期機能

SK2929 opened the pull request:
# WAHT
１、メッセージ送信機能の非同期通信化
(Lesson10 機能を充実させる)
# WHY
投稿機能の非同期機能を作成できるようになるため

Commit:
first commit

Commit:
Edit.gitignore

# 末尾に次の記述を追加
public/uploads/*

Commit:
edit..gitignore

JSを保存できるように、記述を削除

Commit:
edit.gitignore

余計なファイルが生成されないように編集

Commit:
Create.messages.js

3. フォームが送信されたら、イベントが発火するようにする
console.log(this)で値が取得できるか検証

Commit:
Edit.message.js

4. 2のイベントが発火したときにAjaxを使用して、messages#createが動くようにする

Commit:
Edit-ignore

Commit:
remove log files from git management

Commit:
Git-add-keep

Commit:
Edit-message.js

4. 2のイベントが発火したときにAjaxを使用して、messages#createが動くようにする

・Ajaxを使用して、messages#createを呼ぶ
・必要なAjax関数のオプションを揃えて非同期通信を実装
・console.log(this)記述の削除

Commit:
Edit-messages_controller.rb

5. messages#createでメッセージを保存し、respond_toを使用してHTMLとJSONの場合で処理を分ける

Commit:
Create-create.json.jbuilder

6. jbuilderを使用して、作成したメッセージをJSON形式で返す

メッセージ表示に必要な変数を指定

Commit:
Edit-image_uploader.rb

minimagickを適応させるための記述

Commit:
Edit-message.js

7. 返ってきたJSONをdoneメソッドで受取り、HTMLを作成する
（doneメソッドで受け取る記述は、53行目〜56行目）
（HTMLを作成する記述は2行目〜39行目）

8. 7で作成したHTMLをメッセージ画面の一番下に追加する（55行目）

9. HTMLを追加した分、メッセージ画面を下にスクロールする（57行目）

10. 非同期に失敗した場合の処理も準備する（59、60行目）

ky3090 commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,67 @@
+$(function(){
+  function buildHTML(message){
```
プログラミングにはDRY(Don't repeat yourself)という法則があります。
重複は無駄であるという概念があるので、重複している部分を重複させないようにしましょう！
ヒント
参考演算子

ky3090 commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,67 @@
+    .done(function(data){
+      var html = buildHTML(data);
+      $('.messages').append(html)
+      $('.form__message').val('')
```
フォームの値をリセットさせる記述が欲しいです！

ky3090 reviewed (commented):
logやcacheがコミットされているのはなぜですか？

SK2929 commented:
logやcacheがコミットされていたのは、gitignoreがうまく適応されていなくて、何かを変更する度に余計なファイルが作成されていました。
コネクトやメンターに相談して、現在は解決しています。

Commit:
Edit-message.js

指摘事項の修正
1、DRY(Don't repeat yourself)という法則により、重複している部分を三項演算子で表現（3〜16行目）
2、フォームの値をリセットさせる記述を追加
（36行目、画像フォーム内の画像をリセットする。22行目、SENDボタンの押した時の状態をリセットする。）

SK2929 commented:
指摘されていた２点を修正しました

miyagit reviewed (commented):
修正されたところが治ってないので、全て修正してからお願いします。

Commit:
Edit-message.js

$('.form__submit').removeAttr('data-disable-with’)から
$('.form__submit').prop("disabled", false)へ記述を変更して、フォームの値をリセットさせる動作を行う。

hzak2xx commented:
https://github.com/SK2929/chat-space/pull/2/files#r236222375
jQueryのformのresetを使ってください

Commit:
Edit-message.js

$('.form__message').val('')
$('.form__mask').val('')
から
$('#new_message')[0].reset();
Resetを使用しまとめてリセットさせるように変更

ghost commented:
LGTMです！
